### ISSUE TITLE
pix2pix : Further explanation regarding "combine" model

### ISSUE THREAD

**Issue by nobodykid** [eriklindernoren/Keras-GAN#139]
Hello, I'm currently trying to understand how you used `combined` model, especially in [pix2pix](https://github.com/eriklindernoren/Keras-GAN/blob/master/pix2pix/pix2pix.py).

I need to know, why the `combined` model outputs both the discriminator result (`valid`) AND the generator result (`fake_A`)?

Also, in [line 167](https://github.com/eriklindernoren/Keras-GAN/blob/master/acgan/acgan.py#L167), it said
```
g_loss = self.combined.train_on_batch([imgs_A, imgs_B], [valid, imgs_A])
```
Is this `valid` refers to the `valid` of discriminator output, or the ground truth `valid` used to measure loss of real images?

**Comment by eriklindernoren** [comment]
The combined model outputs both the translated images produced by the generator (measured against the ground truth image using pixelwise mse loss) and the classification by the discriminator (which we want to be labeled as valid for the generator to fool the discriminator). In the line you posted above we use `valid` as ground truth for the discriminator output and `imgs_A` as ground truth for the images outputted by the generator.

**Comment by nobodykid** [comment]
I noticed that you also used two losses in the combined model. Are those two losses correspond to each output in `combined` or they're used on all output?

**Comment by eriklindernoren** [comment]
Yes, each is applied to measure the loss of the model's corresponding output.

**Comment by nobodykid** [comment]
Aah I see

As for the naming of both discriminator output and real-image ground truth, should it be in the same name? Or it was coincidence?
